core/shared/src/main/scala/io/circe/Decoder.scala

```diff
@@ -776,6 +776,11 @@ final object Decoder extends TupleDecoders with ProductDecoders with LowPriority
     new SemigroupK[Decoder] with MonadError[Decoder, DecodingFailure] {
       final def combineK[A](x: Decoder[A], y: Decoder[A]): Decoder[A] = x.or(y)
       final def pure[A](a: A): Decoder[A] = const(a)
+      final override def pureEval[A](a: Eval[A]): Decoder[A] = new Decoder[A] {
+        final def apply(c: HCursor): Result[A] = Xor.right(a.value)
```

Comment:
actually, I guess `a.value` can throw if execution throws an exception. Should this throw or should we use `Xor.catchNonFatal` (or whatever it is).